packages/amplify-e2e-tests/src/__tests__/auth_6.test.ts

```diff
@@ -85,13 +85,28 @@ describe('zero config auth', () => {
 
     // override new env
     await amplifyOverrideAuth(projRoot, {});
-    const srcOverrideFilePath = path.join(__dirname, '..', '..', 'overrides', 'override-auth.ts');
+
+    // this is where we will write our override logic to
     const destOverrideFilePath = path.join(projRoot, 'amplify', 'backend', 'auth', `${authResourceName}`, 'override.ts');
+
+    // test override file in compilation error state
+    const srcInvalidOverrideCompileError = path.join(__dirname, '..', '..', 'overrides', 'override-compile-error.txt');
+    fs.copyFileSync(srcInvalidOverrideCompileError, destOverrideFilePath);
+    await expect(amplifyPushOverride(projRoot)).rejects.toThrowError();
+
+    // test override file in runtime error state
+    const srcInvalidOverrideRuntimeError = path.join(__dirname, '..', '..', 'overrides', 'override-runtime-error.txt');
+    fs.copyFileSync(srcInvalidOverrideRuntimeError, destOverrideFilePath);
+    await expect(amplifyPushOverride(projRoot)).rejects.toThrowError();
+
+    // test happy path
+    const srcOverrideFilePath = path.join(__dirname, '..', '..', 'overrides', 'override-auth.ts');
     fs.copyFileSync(srcOverrideFilePath, destOverrideFilePath);
     await amplifyPushOverride(projRoot);
-    // check overidden config
-    const overridenUserPool = await getUserPool(id, meta.providers.awscloudformation.Region);
-    expect(overridenUserPool.UserPool).toBeDefined();
-    expect(overridenUserPool.UserPool.DeviceConfiguration.ChallengeRequiredOnNewDevice).toBe(true);
+
+    // check overwritten config
+    const overwrittenUserPool = await getUserPool(id, meta.providers.awscloudformation.Region);
+    expect(overwrittenUserPool.UserPool).toBeDefined();
```

Comment:
this is great.
Should we add similar coverage for other places we use overrides? i.e. override api, storage etc - all of them spin up NodeVM separately, so I suspect that testing auth and custom doesn't cover those places.

Reply:
good point; i will add before merging - i think its great idea, just wanted to get this up for review before writing all the tests

Reply:
yeah , Nice catch there !!

Reply:
S3, DynamoDb, & Root Stack coverage has been added

API test logic will be included in data repo